src/context/wizkidsContext.js

```diff
@@ -11,14 +11,52 @@ function Provider({children}) {
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState("");
 
-  
+  const [fetchedPositions, setFetchPositions] = useState([]); //show the available positions for the dropdown fiter
+  const [filteredPosition, setFilteredPosition] = useState("All");
+  const [filteredWizkids, setFilteredWizkids] = useState([]);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  useEffect(() => {
+    setFilteredWizkids(wizkids);
+  }, [wizkids]);
+
+  useEffect(() => {
+    searchWizkids(searchQuery);
+  }, [filteredPosition])
+
+  useEffect(() => {
+    searchWizkids(searchQuery);
+  }, [searchQuery])
+
+  const searchWizkids = () => {
+    console.log(searchQuery)
+    if(filteredPosition === "All") {
+      setFilteredWizkids(wizkids.filter((item) =>
+      item.name.toLowerCase().includes(searchQuery.toLowerCase())
+    ));
+    } else{
+      setFilteredWizkids(wizkids.filter((item) =>
+      item.name.toLowerCase().includes(searchQuery.toLowerCase()) && item.position.toLowerCase().includes(filteredPosition.toLowerCase())
+    ));
```

Comment:
This function can probably be smaller, the first `.includes` happens always, only the last part is optional 

```
let filteredRestults = wizkids.filter((item) => searchQuery stuff

    if(filteredPosition !== "All") {
        filteredRestults = filteredRestults.filter((item) => filteredPosition stuff
    }

    return filteredRestults;
```

Reply:
Something like this could make it shorter for example